### PR TITLE
Add some missing ftdetect files

### PR DESCRIPTION
**Pull request by tssm**
I started adding Nix and Fish, which I use, but I decided to add some others that I didn't find in the Neovim runtime by looking at how support plugins do it

### CODE REVIEW

File: ftdetect/julia.vim [hunk 1]
```diff
@@ -0,0 +1 @@
+autocmd BufRead,BufNewFile *.jl setfiletype=julia
```
Comment (bfredl) [inline ftdetect/julia.vim:1]: hmm I think it needs to be `setfiletype julia`

Comment (tssm) [reply ftdetect/julia.vim]: Right, sorry, this is why I should do this kind of things during the week. BTW, the other files use `setlocal filetype` instead. Maybe that's a better option? My understanding is `setfiletype` only works if the file type has not been set already, which seems nicer, but I'm not sure, maybe it doesn't matter

**Review by vigoux — APPROVED** [review]
Let's also wait for @stsewd and @theHamsta opinion on this

**Comment by theHamsta** [comment]
I don't have an opinion on filetpypes :smile:

**Review by theHamsta — APPROVED** [review]
Looks good!

**Review by stsewd — APPROVED** [review]
Thanks!

**Pull request merged**